feat(elb): update elb listener resource for new param (#3991)
huaweicloud/terraform-provider-huaweicloud: docs/resources/elb_listener.md

* `protocol` - (Required, String, ForceNew) The protocol can either be TCP, UDP, HTTP or HTTPS. Changing this creates a
  new listener.

* `loadbalancer_id` - (Required, String, ForceNew) The load balancer on which to provision this listener. Changing this
  creates a new listener.

* `protocol_port` - (Optional, Int, ForceNew) The port used by the listener. The **QUIC** listener port cannot be 4789

>   or the same as the **UDP** listener port. If this parameter is set to 0, `port_ranges` is required.

Reviewer: ```suggestion
* `protocol_port` - (Optional, Int, ForceNew) The port used by the listener. The **QUIC** listener port cannot be **4789**
  or the same as the **UDP** listener port. If this parameter is set to **0**, `port_ranges` is required. Changing this creates a new listener.
```

Author: done

* `tags` - (Optional, Map) The key/value pairs to associate with the listener.

* `gzip_enable` - (Optional, Bool) Whether to enable gzip compression for a load balancer. The default value is **false**.
  This parameter can be configured only for **HTTP**, **HTTPS**, and **QUIC** listeners.

* `port_ranges` - (Optional, List, ForceNew) The port monitoring range (closed range), specify up to 10 port groups, each
  group range must not overlap. This field can only be passed in when protocol_port is 0 or protocol_port is not passed
  in. Only **TCP**, **UDP**, and **TCPSSL** listening support this field.

>   The [port_ranges](#ELB_port_ranges) structure is documented below.

Reviewer: ```suggestion
* `port_ranges` - (Optional, List, ForceNew) The port monitoring range (closed range), specify up to 10 port groups, each
  group range must not overlap. This field can only be passed in when `protocol_port` is **0** or empty. Only **TCP**, **UDP**, and **TCPSSL** listening support this field.
  The [port_ranges](#ELB_port_ranges) structure is documented below.
  Changing this creates a new listener.
```
put this param together with protocol_port

Author: done

<a name="ELB_port_ranges"></a>
The `port_ranges` block supports:

* `start_port` - (Optional, Int, ForceNew) Specifies the start port.

> * `end_port` - (Optional, Int, ForceNew) Specifies the end port.

Reviewer: ```suggestion
* `start_port` - (Required, Int, ForceNew) Specifies the start port. Changing this creates a new listener.

* `end_port` - (Required, Int, ForceNew) Specifies the end port. Changing this creates a new listener.
```

Author: done
